### ISSUE TITLE
Search user experience is not accessible

### ISSUE THREAD

**Issue by lfdebrux** [alphagov/tech-docs-gem#262]
From a recent DAC audit:

> The main page region appears to be hidden from assistive technology when the dialog is triggered. For sighted users, the dialog obscures the main page region, and this acts as a prompt to close it. However, this dialog is non-modal so for screen reader users, this content appears as if it was simply inset on the page after the search button. This means that users navigating with screen reading software may not make the association between needing to close the dialog in order to access the main page content again.
>
> This region is marked up as a dialog with a clear label, however, as focus is not trapped it may not be easy for screen reader users to determine the boundaries of the dialog. When a search is explicitly submitted, focus is taken to the dialog, but lands on the ‘Offer GovWifi in your public sector organisation’ link, so that users either have to tab backwards to close it or otherwise tab through the page. Screen reader users may not be aware that the ‘close’ button is present, as they are navigated straight past it, and as much of the surrounding page appears to be available, would not necessarily infer the need to ‘close’ it.
>
> Additionally, the results dialog appears as a user types in the search field without being explicitly requested. This means that typing within the edit field has the unintended direct consequence of hiding the main page content. Further, unless the form is explicitly submitted, screen reader users are not informed that the page has updated. This means that users who type in the field mistakenly may struggle to infer why the main content has now become hidden.
>
> WCAG 2.1 (A)
>   - 3.2.2 On Input - https://www.w3.org/WAI/WCAG21/Understanding/on-input.html
>   - 2.4.3 Focus Order - https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html

---

There are a couple of possible solutions to this:

1. We could go with the solution outlined by DAC and make the search results act more like a modal dialog, however as this [comment by @36degrees suggests this might not be a good model](https://github.com/alphagov/tech-docs-gem/pull/239#issuecomment-872992709).

2. @colinbm has raised PR #239 which removes the live updating, leaves the content visible, and reduces the content's opacity when the search results are visible, which may be an improvement from an accessibility standpoint but needs refinement and testing with a screenreader.

3. Make the search results behave less like a dialog and instead they just push down the rest of the content, as suggested by @selfthinker.

4. We could remove the live search functionality altogether - so no search suggestions, and no search dialog, just a normal form submission and a separate search results page, as suggested by @m-green. @zapthedingbat has raised PR #259 which does part of this, but creating a separate search results page would probably be a weeks worth extra work.

Option 4 is probably the most accessible and most guaranteed to work. It was also suggested by @selfthinker at the GDS accessibility clinic.

---

> NOTE: We want to resolve this issue by September 30th.
